### DISCUSSION TITLE
The way to enable WarningAsError on UnusedIncludes

### DISCUSSION THREAD

**Discussion by cacosub7** [clangd/clangd#1580] [Q&A] [answered]
I am using the --check mode of clangd for CI checks. The known implementation is that clangd only exposes error-level problems in this mode. Based on the information from this[ issue](https://github.com/clangd/clangd/issues/1447), I can currently expose the checks I need by using the WarningAsError option in `.clang-tidy` and Add `-Werror=Flag` in `.clangd`. However, the `UnusedIncludes: Strict` option provided by clangd is very useful, and I want to add it to my CI checks. But it seems to be a Warn-level check. How can I configure `.clangd` and `.clang-tidy` to apply the WarningAsError feature to 'UnusedIncludes'? Due to project CI environment restrictions, I cannot deploy it by modifying the source code of clangd for now.

The information collected so far:
1. misc-unused-include option was found on [LLVM Review](https://reviews.llvm.org/D122677?id=418946#:~:text=CheckFactories.registerCheck%3CUnusedIncludesCheck%3E(%22misc%2Dunused%2Dincludes%22)%3B), but no related release information was found on GitHub.
2. As mentioned earlier, I saw the decoupled `clang-include-cleaner` check tool, but there was not enough information for me to run it.

**Comment by HighCommander4** [comment, marked as answer]
> How can I configure `.clangd` and `.clang-tidy` to apply the WarningAsError feature to 'UnusedIncludes'?

I don't think this is currently possible without code changes.

**Reply by cacosub7** [reply]
Got it :).
